### DISCUSSION TITLE
.\compile\install.bat执行失败

### DISCUSSION THREAD

**Discussion by samoyedsun** [ejoy/ant#56] [Q&A]
E:\luamake>
E:\luamake>
E:\luamake>.\compile\install.bat
此时不应有 GOTO。
E:\luamake>

**Comment by actboy168** [comment]
你的操作系统版本，以及你的编译器版本？

**Reply by samoyedsun** [reply]
Windows 11 专业版 22H2, 22621.3007, 64 位操作系统, 基于 x64 的处理器
Microsoft Visual Studio Community 2022 (64 位) - Current
版本 17.4.2
Microsoft.NET Framework
版本 4.8.09032

**Reply by actboy168** [reply]
在luamake中将echo打开，然后再执行.\compile\install.bat，并把输出的信息发上来看看。
``` diff
diff --git a/compile/build.bat b/compile/build.bat
index 82e3c12..dab09bc 100644
--- a/compile/build.bat
+++ b/compile/build.bat
@@ -1,6 +1,5 @@
-@ECHO OFF
+
 CALL compile\msvc\find_msvc.bat
 CALL compile\msvc\generate_msvc_deps_prefix.bat
-@ECHO ON

 ninja -f build\msvc\compile.ninja %*
```

**Reply by samoyedsun** [reply]
E:\luamake>.\compile\install.bat

E:\luamake>ECHO OFF
系统找不到指定的路径。

E:\luamake>ninja -f build\msvc\compile.ninja
[1/42] Compile C++ build/msvc/obj/source_bee/socket.obj
FAILED: build/msvc/obj/source_bee/socket.obj
cl /nologo /showIncludes -c bee.lua/bee/net/socket.cpp /Fobuild/msvc/obj/source_bee/socket.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua/bee/net/socket.cpp(3): fatal error C1083: 无法打开包括文件: “winsock2.h”: No such file or directory
[2/42] Compile C++ build/msvc/obj/source_bee/format.obj
FAILED: build/msvc/obj/source_bee/format.obj
cl /nologo /showIncludes -c bee.lua/3rd/fmt/format.cc /Fobuild/msvc/obj/source_bee/format.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
E:\luamake\bee.lua\3rd\fmt\fmt/format-inl.h(11): fatal error C1034: algorithm: 不包括路径集
[3/42] Compile C++ build/msvc/obj/source_bee/module_version_win.obj
FAILED: build/msvc/obj/source_bee/module_version_win.obj
cl /nologo /showIncludes -c bee.lua/bee/platform/win/module_version_win.cpp /Fobuild/msvc/obj/source_bee/module_version_win.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua/bee/platform/win/module_version_win.cpp(1): fatal error C1083: 无法打开包括文件: “Windows.h”: No such file or directory
[4/42] Compile C++ build/msvc/obj/source_bee/unlink_win.obj
FAILED: build/msvc/obj/source_bee/unlink_win.obj
cl /nologo /showIncludes -c bee.lua/bee/platform/win/unlink_win.cpp /Fobuild/msvc/obj/source_bee/unlink_win.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua/bee/platform/win/unlink_win.cpp(1): fatal error C1083: 无法打开包括文件: “Windows.h”: No such file or directory
[5/42] Compile C++ build/msvc/obj/source_bee/file_handle_win.obj
FAILED: build/msvc/obj/source_bee/file_handle_win.obj
cl /nologo /showIncludes -c bee.lua/bee/utility/file_handle_win.cpp /Fobuild/msvc/obj/source_bee/file_handle_win.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua/bee/utility/file_handle_win.cpp(1): fatal error C1083: 无法打开包括文件: “Windows.h”: No such file or directory
[6/42] Compile C++ build/msvc/obj/source_bee/setname.obj
FAILED: build/msvc/obj/source_bee/setname.obj
cl /nologo /showIncludes -c bee.lua/bee/thread/setname.cpp /Fobuild/msvc/obj/source_bee/setname.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua\bee/utility/zstring_view.h(3): fatal error C1083: 无法打开包括文件: “cassert”: No such file or directory
[7/42] Compile C   build/msvc/obj/source_bee/lua-seri.obj
FAILED: build/msvc/obj/source_bee/lua-seri.obj
cl /nologo /showIncludes -c bee.lua/3rd/lua-seri/lua-seri.c /Fobuild/msvc/obj/source_bee/lua-seri.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Itools/lua54 /D_WIN32_WINNT=0x0601 /DNDEBUG /wd4244 /utf-8 /std:c11
tools/lua54\lua.h(12): fatal error C1083: 无法打开包括文件: “stdarg.h”: No such file or directory
[8/42] Compile C++ build/msvc/obj/source_bee/process_select.obj
FAILED: build/msvc/obj/source_bee/process_select.obj
cl /nologo /showIncludes -c bee.lua/bee/subprocess/process_select.cpp /Fobuild/msvc/obj/source_bee/process_select.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua\bee/utility/dynarray.h(3): fatal error C1083: 无法打开包括文件: “memory.h”: No such file or directory
[9/42] Compile C++ build/msvc/obj/source_bee/endpoint.obj
FAILED: build/msvc/obj/source_bee/endpoint.obj
cl /nologo /showIncludes -c bee.lua/bee/net/endpoint.cpp /Fobuild/msvc/obj/source_bee/endpoint.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua\bee/utility/zstring_view.h(3): fatal error C1083: 无法打开包括文件: “cassert”: No such file or directory
[10/42] Compile C++ build/msvc/obj/source_bee/file_handle.obj
FAILED: build/msvc/obj/source_bee/file_handle.obj
cl /nologo /showIncludes -c bee.lua/bee/utility/file_handle.cpp /Fobuild/msvc/obj/source_bee/file_handle.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua\bee/utility/file_handle.h(4): fatal error C1083: 无法打开包括文件: “cstdint”: No such file or directory
[11/42] Compile C++ build/msvc/obj/source_bee/unicode_win.obj
FAILED: build/msvc/obj/source_bee/unicode_win.obj
cl /nologo /showIncludes -c bee.lua/bee/platform/win/unicode_win.cpp /Fobuild/msvc/obj/source_bee/unicode_win.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua/bee/platform/win/unicode_win.cpp(1): fatal error C1083: 无法打开包括文件: “Windows.h”: No such file or directory
[12/42] Compile C++ build/msvc/obj/source_bee/subprocess_win.obj
FAILED: build/msvc/obj/source_bee/subprocess_win.obj
cl /nologo /showIncludes -c bee.lua/bee/subprocess/subprocess_win.cpp /Fobuild/msvc/obj/source_bee/subprocess_win.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua/bee/subprocess/subprocess_win.cpp(1): fatal error C1083: 无法打开包括文件: “Shobjidl.h”: No such file or directory
[13/42] Compile C++ build/msvc/obj/source_bee/filewatch_win.obj
FAILED: build/msvc/obj/source_bee/filewatch_win.obj
cl /nologo /showIncludes -c bee.lua/bee/filewatch/filewatch_win.cpp /Fobuild/msvc/obj/source_bee/filewatch_win.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-bee.lua/bee/filewatch/filewatch_win.cpp(1): fatal error C1083: 无法打开包括文件: “Windows.h”: No such file or directory
[14/42] Compile C++ build/msvc/obj/source_bee/simplethread_win.obj
FAILED: build/msvc/obj/source_bee/simplethread_win.obj
cl /nologo /showIncludes -c bee.lua/bee/thread/simplethread_win.cpp /Fobuild/msvc/obj/source_bee/simplethread_win.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua/bee/thread/simplethread_win.cpp(1): fatal error C1083: 无法打开包括文件: “Windows.h”: No such file or directory
[15/42] Compile C++ build/msvc/obj/source_bee/version.obj
FAILED: build/msvc/obj/source_bee/version.obj
cl /nologo /showIncludes -c bee.lua/bee/platform/version.cpp /Fobuild/msvc/obj/source_bee/version.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua\bee/platform/version.h(3): fatal error C1083: 无法打开包括文件: “cstdint”: No such file or directory
[16/42] Compile C++ build/msvc/obj/source_bee/path_helper.obj
FAILED: build/msvc/obj/source_bee/path_helper.obj
cl /nologo /showIncludes -c bee.lua/bee/utility/path_helper.cpp /Fobuild/msvc/obj/source_bee/path_helper.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua\bee/error.h(3): fatal error C1083: 无法打开包括文件: “string”: No such file or directory
[17/42] Compile C++ build/msvc/obj/source_bee/lua_filesystem.obj
FAILED: build/msvc/obj/source_bee/lua_filesystem.obj
cl /nologo /showIncludes -c bee.lua/binding/lua_filesystem.cpp /Fobuild/msvc/obj/source_bee/lua_filesystem.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /Itools/lua54 /DBEE_STATIC /D_CRT_SECURE_NO_WARNINGS /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua\bee/error.h(3): fatal error C1083: 无法打开包括文件: “string”: No such file or directory
[18/42] Compile C++ build/msvc/obj/source_bee/atomic_semaphore.obj
FAILED: build/msvc/obj/source_bee/atomic_semaphore.obj
cl /nologo /showIncludes -c bee.lua/bee/thread/atomic_semaphore.cpp /Fobuild/msvc/obj/source_bee/atomic_semaphore.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua\bee/thread/atomic_semaphore.h(3): fatal error C1083: 无法打开包括文件: “atomic”: No such file or directory
[19/42] Compile C++ build/msvc/obj/source_bee/error.obj
FAILED: build/msvc/obj/source_bee/error.obj
cl /nologo /showIncludes -c bee.lua/bee/error.cpp /Fobuild/msvc/obj/source_bee/error.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua\bee/error.h(3): fatal error C1083: 无法打开包括文件: “string”: No such file or directory
[20/42] Compile C++ build/msvc/obj/source_bee/spinlock.obj
FAILED: build/msvc/obj/source_bee/spinlock.obj
cl /nologo /showIncludes -c bee.lua/bee/thread/spinlock.cpp /Fobuild/msvc/obj/source_bee/spinlock.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua\bee/nonstd/atomic_flag.h(3): fatal error C1083: 无法打开包括文件: “atomic”: No such file or directory
[21/42] Compile C++ build/msvc/obj/source_bee/lua_platform.obj
FAILED: build/msvc/obj/source_bee/lua_platform.obj
cl /nologo /showIncludes -c bee.lua/binding/lua_platform.cpp /Fobuild/msvc/obj/source_bee/lua_platform.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /Itools/lua54 /DBEE_STATIC /D_CRT_SECURE_NO_WARNINGS /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua\bee/error.h(3): fatal error C1083: 无法打开包括文件: “string”: No such file or directory
[22/42] Compile C++ build/msvc/obj/source_bee/lua_filewatch.obj
FAILED: build/msvc/obj/source_bee/lua_filewatch.obj
cl /nologo /showIncludes -c bee.lua/binding/lua_filewatch.cpp /Fobuild/msvc/obj/source_bee/lua_filewatch.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /Itools/lua54 /DBEE_STATIC /D_CRT_SECURE_NO_WARNINGS /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua\bee/error.h(3): fatal error C1083: 无法打开包括文件: “string”: No such file or directory
ninja: build stopped: subcommand failed.

E:\luamake>

**Reply by actboy168** [reply]
看起来你没有正确的修改build.bat，修改后的build.bat应该这样
``` bat
 CALL compile\msvc\find_msvc.bat
 CALL compile\msvc\generate_msvc_deps_prefix.bat
 ninja -f build\msvc\compile.ninja %*
```

**Reply by samoyedsun** [reply]
修改后又执行是下面这样

```bat
E:\luamake>.\compile\install.bat

E:\luamake>CALL compile\msvc\find_msvc.bat

E:\luamake>IF NOT "" == "" GOTO QUIT

E:\luamake>IF /I "AMD64" == "AMD64" GOTO AMD64

E:\luamake>SET ProgramFiles=C:\Program Files (x86)

E:\luamake>SET ARCH=x64

E:\luamake>FOR /F "usebackq tokens=*" %i in (`"C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" -latest -prerelease -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationPath`) do (SET InstallDir=%i )

E:\luamake>CALL compile\msvc\find_winsdk.bat
系统找不到指定的路径。
[1/42] Compile C++ build/msvc/obj/source_bee/spinlock.obj
FAILED: build/msvc/obj/source_bee/spinlock.obj
cl /nologo /showIncludes -c bee.lua/bee/thread/spinlock.cpp /Fobuild/msvc/obj/source_bee/spinlock.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua\bee/nonstd/atomic_flag.h(3): fatal error C1083: 无法打开包括文件: “atomic”: No such file or directory
[2/42] Compile C++ build/msvc/obj/source_bee/setname.obj
FAILED: build/msvc/obj/source_bee/setname.obj
cl /nologo /showIncludes -c bee.lua/bee/thread/setname.cpp /Fobuild/msvc/obj/source_bee/setname.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua\bee/utility/zstring_view.h(3): fatal error C1083: 无法打开包括文件: “cassert”: No such file or directory
[3/42] Compile C++ build/msvc/obj/source_bee/module_versio
```

**Reply by actboy168** [reply]
看起来是没找到WindowsSDK,把comiple/msvc/find_winsdk.bat的第一行`@echo off`删了，再执行一次。

**Reply by samoyedsun** [reply]
```bat
E:\luamake>
E:\luamake>
E:\luamake>
E:\luamake>
E:\luamake>.\compile\install.bat

E:\luamake>CALL compile\msvc\find_msvc.bat

E:\luamake>IF NOT "" == "" GOTO QUIT

E:\luamake>IF /I "AMD64" == "AMD64" GOTO AMD64

E:\luamake>SET ProgramFiles=C:\Program Files (x86)

E:\luamake>SET ARCH=x64

E:\luamake>FOR /F "usebackq tokens=*" %i in (`"C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" -latest -prerelease -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationPath`) do (SET InstallDir=%i )

E:\luamake>CALL compile\msvc\find_winsdk.bat

E:\luamake>call :GetWin10SDKVersion HKLM\SOFTWARE\Wow6432Node

E:\luamake>for /F "tokens=1,2*" %i in ('reg query "HKLM\SOFTWARE\Wow6432Node\Microsoft\Microsoft SDKs\Windows\v10.0" /v "InstallationFolder"') do (if "%i" == "InstallationFolder" (set WindowsSdkDir=%~k ) )

E:\luamake>(if "HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Microsoft\Microsoft" == "InstallationFolder" (set WindowsSdkDir= ) )

E:\luamake>(if "InstallationFolder" == "InstallationFolder" (set WindowsSdkDir=D:\Windows Kits\10\ ) )

E:\luamake>setlocal enableDelayedExpansion

E:\luamake>if not "D:\Windows Kits\10\" == "" for /F %i IN ('dir "D:\Windows Kits\10\include\" /b /ad-h /on') DO (if exist "D:\Windows Kits\10\include\%i\um\winsdkver.h" (
set result=%i
 if "!result:~0,3!" == "10." (set SDK=!result! )
) )

E:\luamake>(if exist "D:\Windows Kits\10\include\10.0.18362.0\um\winsdkver.h" (
set result=10.0.18362.0
 if "!result:~0,3!" == "10." (set SDK=!result! )
) )

E:\luamake>(if exist "D:\Windows Kits\10\include\10.0.20348.0\um\winsdkver.h" (
set result=10.0.20348.0
 if "!result:~0,3!" == "10." (set SDK=!result! )
) )

E:\luamake>(if exist "D:\Windows Kits\10\include\10.0.22000.0\um\winsdkver.h" (
set result=10.0.22000.0
 if "!result:~0,3!" == "10." (set SDK=!result! )
) )

E:\luamake>endlocal   & set WindowsSDKVersion=10.0.22000.0

E:\luamake>exit /B 0

E:\luamake>if errorlevel 1 call :GetWin10SDKVersion HKCU\SOFTWARE\Wow6432Node

E:\luamake>if errorlevel 1 call :GetWin10SDKVersion HKLM\SOFTWARE

E:\luamake>if errorlevel 1 call :GetWin10SDKVersion HKCU\SOFTWARE

E:\luamake>if errorlevel 1 exit /B 1

E:\luamake>exit /B 0

E:\luamake>SET VSCMD_SKIP_SENDTELEMETRY=1

E:\luamake>CALL "\Common7\Tools\vsdevcmd.bat" -arch=x64 -host_arch=x64 -winsdk=10.0.22000.0  1>nul
系统找不到指定的路径。

E:\luamake>CALL compile\msvc\generate_msvc_deps_prefix.bat

E:\luamake>mkdir build\msvc  2>nul

E:\luamake>cd build\msvc

E:\luamake\build\msvc>mkdir temp  2>nul

E:\luamake\build\msvc>type nul 1>temp/test.h

E:\luamake\build\msvc>echo #include "test.h"  1>temp/test.c

E:\luamake\build\msvc>cd temp

E:\luamake\build\msvc\temp>for /F "usebackq delims=: tokens=1,2 skip=1" %G in (`cl /showIncludes /nologo test.c`) do (
set MsvcDepsPrefix=%G:%H:
 goto Break
)

E:\luamake\build\msvc\temp>(
set MsvcDepsPrefix=注意: 包含文件:
 goto Break
)

E:\luamake\build\msvc\temp>cd ..

E:\luamake\build\msvc>rmdir /q /s temp

E:\luamake\build\msvc>echo msvc_deps_prefix = 注意: 包含文件:  1>compile.ninja

E:\luamake\build\msvc>echo subninja compile/ninja/msvc.ninja  1>>compile.ninja

E:\luamake\build\msvc>cd ../..

E:\luamake>ninja -f build\msvc\compile.ninja
[1/42] Compile C   build/msvc/obj/source_bee/lua-seri.obj
FAILED: build/msvc/obj/source_bee/lua-seri.obj
cl /nologo /showIncludes -c bee.lua/3rd/lua-seri/lua-seri.c /Fobuild/msvc/obj/source_bee/lua-seri.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Itools/lua54 /D_WIN32_WINNT=0x0601 /DNDEBUG /wd4244 /utf-8 /std:c11
tools/lua54\lua.h(12): fatal error C1083: 无法打开包括文件: “stdarg.h”: No such file or directory
[2/42] Compile C++ build/msvc/obj/source_bee/file_handle.obj
FAILED: build/msvc/obj/source_bee/file_handle.obj
cl /nologo /showIncludes -c bee.lua/bee/utility/file_handle.cpp /Fobuild/msvc/obj/source_bee/file_handle.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua\bee/utility/file_handle.h(4): fatal error C1083: 无法打开包括文件: “cstdint”: No such file or directory
[3/42] Compile C++ build/msvc/obj/source_bee/unlink_win.obj
FAILED: build/msvc/obj/source_bee/unlink_win.obj
cl /nologo /showIncludes -c bee.lua/bee/platform/win/unlink_win.cpp /Fobuild/msvc/obj/source_bee/unlink_win.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua/bee/platform/win/unlink_win.cpp(1): fatal error C1083: 无法打开包括文件: “Windows.h”: No such file or directory
[4/42] Compile C++ build/msvc/obj/source_bee/version.obj
FAILED: build/msvc/obj/source_bee/version.obj
cl /nologo /showIncludes -c bee.lua/bee/platform/version.cpp /Fobuild/msvc/obj/source_bee/version.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibee.lua /D_WIN32_WINNT=0x0601 /DNDEBUG /utf-8 /std:c++17 /GR-
bee.lua\bee/platform/version.h(3): fatal error C1083: 无法打开
```

**Reply by actboy168** [reply]
你cmd里执行这个的结果是什么？
``` bat
"C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" -latest -prerelease -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -format json
```

**Reply by samoyedsun** [reply]
```bat
E:\luamake>
E:\luamake>"C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" -latest -prerelease -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -format json
[]

E:\luamake>"C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" -latest -prerelease -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -format json
[]

E:\luamake>
```

**Reply by samoyedsun** [reply]
我在公司里的win10上编了下，还是挺顺利；家里这个win11就有这种问题

**Reply by actboy168** [reply]
试试
``` bat
"C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" -all -prerelease -products * -format json
```

**Reply by samoyedsun** [reply]
```bat
E:\luamake>
E:\luamake>
E:\luamake>"C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" -all -prerelease -products * -format json
[]

E:\luamake>"C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" -all -prerelease -products * -format json
[]

E:\luamake>
```

**Reply by actboy168** [reply]
看起来是你的msvc没有正确安装

**Reply by Libnezz** [reply]
只能在macos系统下编译吗
